### DISCUSSION TITLE
Bundle api directory in `dist`

### DISCUSSION THREAD

**Discussion by rtritto** [vikejs/vike#2025] [Discussion]
```console
├── server
│   ├── index.ts         // entrypoint of server
│   ├── api              // directory for routes
│   │   ├── user
│   │   │   └── [id].ts  // GET /user/:id
```

Enable HMR:
```ts
await globalThis.__vikeNode.viteDevServer.ssrLoadModule('/server/api/user/[id].ts', { fixStacktrace: true })
```

How can I generate the route file in `dist` directory?

Expected something like `/dist/api/user/[id].mjs`.

**Comment by rtritto** [comment]
Achieved by https://github.com/vikejs/vike-node/issues/35

**Comment by brillout** [comment]
Response [here](https://github.com/vikejs/vike-node/issues/35#issuecomment-2552280863).

**Comment by brillout** [comment]
@rtritto Please don't duplicate posts. Keep the conversation in one thread. Thank you.
